SwiftGenerics.playground/Contents.swift

```diff
@@ -0,0 +1,99 @@
+import UIKit
+
+struct CountedSet<Element: Hashable>: ExpressibleByArrayLiteral {
+    
+    private(set) var elements = [Element: Int]()
+    
+    init(arrayLiteral elements: Element...) {
+        for element in elements {
+            insert(element: element)
+        }
+    }
+    
+    mutating func insert(element: Element) {
+        if let _ = elements[element] {
+            elements[element]! += 1
```

Comment:
This can be optimized, since we essentially need to do the dictionary lookup twice.
```suggestion
        if let previousCount = elements[element] {
            elements[element] = previousCount + 1
```